framework/doc/content/source/problems/EigenProblem.md

```diff
@@ -2,16 +2,109 @@
 
 !syntax description /Problem/EigenProblem
 
-The `EigenProblem` holds the nonlinear eigen system and organizes the computation
-of the residual and Jacobian in the context of an eigen system.
+`EigenProblem` is the primary eigenvalue system in the MOOSE framework.
+It offers linear and nonlinear eigensolvers by leveraging [SLEPc](https://slepc.upv.es)
+capabilities (mainly the EPS system). For linear problems,
+the system supports one eigenvalue and multiple eigenvalues. For
+nonlinear problems, we currently support solving for the smallest (in magnitude)
+eigenvalue. The development of nonlinear eigensolver was mainly motivated
+by neutron transport criticality calculations, but the implementation
+and algorithm are general and applicable to other applications.
 
-It also takes care of the following customizations of the solve:
+A generalized (linear) eigenvalue problem can uniformly be rewritten as
+\begin{equation}
+Ax = \lambda Bx,
+\end{equation}
+$A$ and $B$ are matrices, $x$ is an eigenvector, and $\lambda$ is the eigenvalue.
+Generally speaking, $A$ is a nonsingular matrix, and $B$ can be singular
+(e.g., an eigenvalue problem with Dirichlet boundary conditions).
+Many solvers in SLEPc can handle this linear eigenproblem.
+We can choose these algorithms in the executioner block.
+Here is an example with Jacobi Davidson:
+
+!listing test/tests/problems/eigen_problem/eigensolvers/gipm.i
+         block=Executioner
+         link=False
+
+Note that to use the linear solvers from SLEPc, we need to form accurate
+$A$ and $B$ that will be utilized to represent the underlying physics.
+If $A$ and $B$ are incorrect, you will likely end up with the
+wrong eigenvalue and eigenvector. For that reason, if you
+want to use linear eigenvalue solvers, you need to implement
+Jacobian by hand or with AD in the corresponding kernels.
+Details of all the linear eigensolvers can be found in
+[SLEPc Users Manual](https://slepc.upv.es/documentation/slepc.pdf).
+
+In a multiphysics environment, physics are often nonlinear. For instance,
+ neutron transport with temperature feedback. In this case,
+ $A$ and $B$ are not constant anymore; instead, they depend on
+ the eigenvector directly or indirectly. To describe our nonlinear
+ solver, we write a generalized nonlinear eigenvalue problem as,
+\begin{equation}
+A(x) = \lambda B(x),
+\end{equation}
+where $A(\cdot)$ and $B$ are nonlinear functions. In the EigenProblem
+system, the kernels of $A(\cdot)$ are referred to as "noneigen,"
+and the kernels of $B(\cdot)$ are referred to as "eigen."
+To utilize Newton to solve this problem, we define $\lambda = \frac{1}{|B(x)|}$,
```

Comment:
Newton => Newton method (to differentiate Newton solve type).

Reply:
we define => we let.

Also add: this condition serves as a normalization condition as well to ensure the uniqueness of the eigenvector.